src/rgw/rgw_main.cc

```diff
@@ -78,9 +78,9 @@ void signal_shutdown()
     int val = 0;
     int ret = write(signal_fd[0], (char *)&val, sizeof(val));
     if (ret < 0) {
-      int err = -errno;
+      ret = -errno;
       derr << "ERROR: " << __func__ << ": write() returned "
-           << cpp_strerror(-err) << dendl;
+           << cpp_strerror(ret) << dendl;
```

Comment:
actually there is no need ot change the sign of ret here,  cpp_strerror already has an internal check for this purpose, so just call cpp_strerror(errno);